src/core/models/model_list.py

```diff
@@ -3,15 +3,68 @@
 from core.models.model import ActuatorModel, EnvironmentModel, SensorModel
 from core.state import State, array_to_state
 from core.config import Config
-from utils.constants import ModelEnum, State_Type
+from utils.constants import BodyEnum, ModelEnum, State_Type
+from utils.astropy_util import get_ephemeris
+from astropy.constants import G, M_earth, M_sun
 
 
 class AttitudeDynamics(EnvironmentModel):
     ...
 
 
 class PositionDynamics(EnvironmentModel):
-    ...
+    """
+    The position dynamics model implementation.
+    """
+
+    def __init__(self, parameters) -> None:
+        super().__init__(parameters)
+
+    def evaluate(self, t: float, state: State) -> Dict[str, State_Type]:
+        return super().evaluate(t, state)
+
+    def d_state(self, t: float, state: State) -> Dict[str, State_Type]:
+        """
+        Takes the derivative of a vector [r v] to compute [v a], where r is a position vector,
+        v is the velocity vector, and a is the acceleration vector
+        Args:
+            t (float): the initial time
+            state (State): the initial state
+
+        Returns:
+            Dict[str, State_Type]: The updated vector [v a]
+        """
+        # position vectors from moon/sun/earth/craft to the origin, where the origin is the Earth's center of mass
+        r_co = np.array([state.x, state.y, state.z])
+        r_mo = np.array(get_ephemeris(t, BodyEnum.Moon))
+        r_so = np.array(get_ephemeris(t, BodyEnum.Sun))
+        r_eo = np.array((0.0, 0.0, 0.0))  # Earth is at the origin in GCRS
+
+        # position vectors from body to the craft
+        r_mc = np.subtract(r_mo, r_co)
+        r_sc = np.subtract(r_so, r_co)
+        r_ec = np.subtract(r_eo, r_co)
```

Comment:
Please add comments describing what each of these vectors are (and the fact that they're column vectors in the first place)

Reply:
Agreed, resolved.